### PR TITLE
MakerDAO Governance-Module Update

### PR DESCRIPTION
**Pull request by 0xtaf**
# Agent name here

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] New Agent
- [ ] Fix an agent error
- [x] Update an agent logic/test/documentation
- [ ] Code style update (formatting, renaming)
- [ ] Other (please describe):

## New Agent PR readiness

- [x] `npm install` command works (i.e. package-lock is updated properly)
- [x] The `name` and `description` fields of `package.json` describe the agent properly
- [x] All the libraries listed in `package.json` are being used
- [x] All the development libraries are in `devDependencies` section of `package.json`
- [x] The agent README is updated
- [x] The agent README sufficiently describes what the agent does
- [x] All the alerts that the agent emits are listed in the README
- [x] The txn/block examples listed in the README generate the expected findings
- [x] Code is easy to follow and/or properly commented
- [x] Agent contains tests
- [x] All tests pass when running `npm test`
- [x] Tests are not making real network calls (i.e. network interaction is properly mocked)
- [x] Tests cover all the paths (i.e. negative & positive test cases)
- [x] When the tests complete, no warning/errors are shown
- [x] `npm start` works without errors when executed using a jsonRpcUrl for each of the networks specified in the README
- [x] The code does what the README describes
- [x] Network calls are properly being cached (if duplicated calls can occur)
- [x] A performance review has been done (e.g. `Promise.all` is used whenever possible)
- [x] The findings fields are filled out correctly
- [x] Findings `protocol` is specified (if the agent is for a specific one)
- [x] `metadata` of the findings contains all the information needed.
- [x] The SDK methods are being used appropriately (e.g. `filterLog`, `filterFunction`, `getEthersProvider`)
- [x] `ABI`s match the contracts used / `args` used from `LogDescriptions` exists in the `ABI`s
- [x] The agent follows Forta recommended [best practices](https://docs.forta.network/en/latest/best-practices/)
- [x] Agent can recover from failed error calls when they can fail (e.g. errors are being caught on calls that can fail)

## Comments (optional)

- Updated forta-agent & forta-agent-tools to the latest versions.
- Changed `createTransactionEvent` function in utils.ts by removing `receipt` field and adding contractAddress field as `null` since it was necessary. 
- Removed everestId fields.
- Ran prettier format

### CODE REVIEW

File: Maker-Agents-Suit-Module/Governance-Module/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "maker-gm",
-  "version": "0.0.1",
+  "version": "0.0.2",
```
Comment (yasmine9) [inline Maker-Agents-Suit-Module/Governance-Module/package.json:3]: Not sure if we should move to version 0.0.2 while the first one wasn't even deployed

**Review by RCantu92 — COMMENTED** [review]
Overall, good work. Left some minor comments

File: Maker-Agents-Suit-Module/Governance-Module/src/agent.ts [hunk 2]
```diff
@@ -1,48 +1,48 @@
...
 
-const spellsManager: AddressManager = 
-  new DeployedAddressesManager(SPELL_DEPLOYER, _web3.eth.getTransactionCount);
-const liftersManager: ListManager = new ListManager(KNOWN_LIFTERS);
-
-export function provideHandleTransaction(addressManager: AddressManager): HandleTransaction {
+export function provideHandleTransaction(
+  chief: string,
+  spellsManager: AddressManager,
+  lifterManager: AddressManager
```
[the remark this answers is not in the document]
Comment (0xtaf) [reply Maker-Agents-Suit-Module/Governance-Module/src/agent.ts:17]: they both have the same Interface of AddressManager so it's not a type but it's an interface which has its isKnownAddress method that's used by both spellsManager and lifterManager

**Review by RCantu92 — APPROVED** [review]
LGTM 🚀

**Review by yasmine9 — CHANGES_REQUESTED** [review]

**Review by RCantu92 — APPROVED** [review]
Left one minor comment, but overall LGTM 🚀

File: Maker-Agents/Governance-Module/src/agent.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  chiefAddress = await chiefFetcher.getChiefAddress("latest");
+};
+
+export const provideHandleTransaction = (
+  chiefAddress: string,
+  spellsManager: AddressManager,
+  lifterManager: AddressManager
+): HandleTransaction => {
+  return async (transactionEvent: TransactionEvent): Promise<Finding[]> => {
+    // Listen to UpdateAddress event & Update the chief contract.
```
[the remark this answers is not in the document]
Comment (0xtaf) [reply Maker-Agents/Governance-Module/src/agent.ts:28]: great catch! I'll delete it as a param.

**Review by yasmine9 — APPROVED** [review]
Updates LGTM ✅

**Pull request merged**